src/data_generators.py

```diff
@@ -43,8 +46,8 @@ def D_data_generator(
         ifrequencies = None
 
     # prepare a zero array the size of the deformation:
-    registration_images = np.zeros((batch_size,) + vol_shape + (1,))
-    adversarial_images = np.zeros((batch_size,) + vol_shape + (1,))
+    registration_images = np.zeros((batch_size,) + tuple(vol_shape) + (1,))
+    adversarial_images = np.zeros((batch_size,) + tuple(vol_shape) + (1,))
```

Comment:
is tuple already with suggested changes in `train_script.py`

Reply:
what do you mean?